src/sdk_codes.ts

```diff
@@ -0,0 +1,171 @@
+  5002: 'SSE event received: ',
+  5003: 'SSE retrying to connect in',
+  5004: 'SSE stopped',
+  // SDK_EVAL_6xxx -
+  6000: 'Evaluation successful: ',
+  6001: 'Evaluation Failed, returning default variation: ',
+  // SDK_METRICS_7xxx
+  7000: 'Metrics thread started with request interval:',
+  7001: 'Metrics stopped',
+  7002: 'Posting metrics failed, reason:',
+  7003: 'Metrics posted successfully',
+};
+
+export function getSDKCodeMessage(key: number): string {
+  if (key in sdkCodes) {
+    return sdkCodes[key];
+  } else {
+    return 'Unknown SDK code';
+  }
+}
+
+export function sdkErrMsg(errorCode: number, appendText = ''): string {
+  return `SDKCODE:${errorCode}: ${getSDKCodeMessage(errorCode)} ${appendText}`;
+}
+
+export function warnMissingSDKKey(logger: Logger): void {
+  logger.warn(sdkErrMsg(1002, ''));
+}
+
+export function infoPollStarted(durationSec: number, logger: Logger): void {
+  logger.info(sdkErrMsg(4000, `${durationSec * 1000}`));
+}
+
+export function infoSDKInitOK(logger: Logger): void {
+  logger.info(sdkErrMsg(1000, ''));
+}
+
+export function infoSDKStartClose(logger: Logger): void {
+  logger.info(sdkErrMsg(3000, ''));
+}
+
+export function infoSDKCloseSuccess(logger: Logger): void {
+  logger.info(sdkErrMsg(3001, ''));
+}
+
+export function infoSDKAuthOK(logger: Logger): void {
+  logger.info(sdkErrMsg(2000, ''));
+}
+
+export function infoPollingStopped(logger: Logger): void {
+  logger.info(sdkErrMsg(4001));
+}
+
+export function infoStreamConnected(logger: Logger): void {
+  logger.info(sdkErrMsg(5000, ''));
+}
+
+export function debugStreamEventReceived(logger: Logger): void {
+  logger.debug(sdkErrMsg(5002));
+}
+
+export function infoStreamStopped(logger: Logger): void {
+  logger.info(sdkErrMsg(5004, ''));
+}
+
+export function infoMetricsThreadStarted(
+  interval: number,
+  logger: Logger,
+): void {
+  logger.info(sdkErrMsg(7000, `${interval}`));
+}
+
+export function infoMetricsSuccess(logger: Logger): void {
+  logger.info(sdkErrMsg(7003, ''));
+}
+
+export function infoMetricsThreadExited(logger: Logger): void {
+  logger.info(sdkErrMsg(7001, ''));
+}
+
+export function debugEvalSuccess(
+  result: string,
+  flagIdentifier: string,
+  target: Target,
+  logger: Logger,
+): void {
+  logger.debug(
+    sdkErrMsg(
+      6000,
+      `result=${result}, flag identifier=${flagIdentifier}, target=${JSON.stringify(
+        target,
+      )}`,
+    ),
+  );
+}
+
+export function warnAuthFailedSrvDefaults(logger: Logger): void {
+  logger.warn(sdkErrMsg(2001, ''));
+}
+
+export function warnFailedInitAuthError(logger: Logger): void {
+  logger.warn(sdkErrMsg(1001, ''));
+}
+
+export function warnAuthFailedExceedRetries(logger: Logger): void {
+  logger.warn(sdkErrMsg(2003, ''));
+}
+
+export function warnAuthRetrying(
+  attempt: number,
+  error: string,
+  logger: Logger,
+): void {
+  logger.warn(
+    sdkErrMsg(2002, `attempt ${attempt}, got error: ${error}, Retrying ...`),
```

Comment:
This will look a bit odd when it outputs `Authentication attempt attempt 3, got error: something, Retrying...`

Reply:
Agreed, the whole string will now be 

`Authentication attempt failed: attempt=3, error=something, continue retrying=true`

Reply:
![Screenshot 2023-06-23 at 12 25 27](https://github.com/harness/ff-nodejs-server-sdk/assets/23323369/c660bcb8-d6fb-4885-9228-db4f0d924545)